src/kafka.c

```diff
@@ -54,6 +54,78 @@ rebalance_cb (rd_kafka_t *rk, rd_kafka_resp_err_t err, rd_kafka_topic_partition_
     }
 }
 
+/*
+ * replace_char
+ *      Replace in string `str` character c with character s.
+ */
+static void replace_char(char *str, char c, char s)
+{
+    while (*str)
+    {
+        if (*str == c)
+            *str = s;
+        ++str;
+    }
+}
+
+static void kafka_set_option(const char *key, const char *value, void *arg)
+{
+    char errstr[512];
+    char buf[128];
+    rd_kafka_conf_t    *conf = (rd_kafka_conf_t *) arg;
+    rd_kafka_conf_res_t res;
+
+    strcpy(buf, key);
```

Comment:
isn't it safer to use `strncpy` here? what if key is longer than 128 characters?
It also makes sense to ensure `buf` is NUL-terminated as the following `replace_char` depends on it.

Reply:
agree